src/logicTemplates/dynamicfee.js

```diff
@@ -0,0 +1,180 @@
+const templates = require('./templates');
+const transaction = require('../transaction');
+const logicSig = require('../logicsig');
+const algosdk = require('../main');
+const nacl = require('../nacl/naclWrappers');
+const address = require('../encoding/address');
+const encoding = require('../encoding/encoding');
+
+class DynamicFee {
+    zeroAddress = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY5HFKQ";
+    /**
+     * DynamicFee contract allows you to create a transaction without
+     * specifying the fee. The fee will be determined at the moment of
+     * transfer.
+     *
+     * Constructor Parameters:
+     * @param {string} receiver: address to receive the assets
+     * @param {int} amount: amount of assets to transfer
+     * @param {int} firstValid: first valid round for the transaction
+     * @param {int} lastValid:  last valid round for the transaction
+     * @param {string} closeRemainder: if you would like to close the account after the transfer, specify the address that would recieve the remainder, else leave undefined
+     * @param {string} lease: leave undefined to generate a random lease, or supply a lease as base64
+     * @returns {DynamicFee}
+     */
+    constructor(receiver, amount, firstValid, lastValid, closeRemainder, lease) {
+        // don't need to validate receiver, closeremainderto - insert will handle that
+        if (!Number.isSafeInteger(amount) || amount < 0) throw Error("amount must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(firstValid) || firstValid < 0) throw Error("firstValid must be a positive number and smaller than 2^53-1");
+        if (!Number.isSafeInteger(lastValid) || lastValid < 0) throw Error("lastValid must be a positive number and smaller than 2^53-1");
+
+        if (closeRemainder === undefined) {
+            closeRemainder = this.zeroAddress
+        }
+        if (lease === undefined) {
+            let leaseBytes = nacl.randomBytes(32);
+            lease = Buffer.from(leaseBytes).toString('base64');
+        }
```

Comment:
Do you mean to explicitly test for only `undefined` here?  Could you do `if (!closeRemainder)` or even set a default value in the constructor signature (where it's readable, of course)?

These are nits :)